tests/Core/Config/SniffListTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Tests for the \PHP_CodeSniffer\Config --sniffs and --exclude arguments.
+ *
+ * @author  Dan Wallis <dan@wallis.nz>
+ * @license https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Tests\Core\Config;
+
+use PHPUnit\Framework\TestCase;
+use PHP_CodeSniffer\Tests\ConfigDouble;
+
+/**
+ * Tests for the \PHP_CodeSniffer\Config --sniffs and --exclude arguments.
+ *
+ * @covers \PHP_CodeSniffer\Config::processLongArgument
+ */
+final class SniffListTest extends TestCase
+{
+
+
+    /**
+     * Ensure that the expected error message is returned for invalid arguments
+     *
+     * @param string $argument sniffs or exclude
+     * @param string $value    list of sniffs to include / exclude
+     * @param string $message  expected error message text
+     *
+     * @return       void
+     * @dataProvider dataInvalidSniffs
+     */
+    public function testInvalid($argument, $value, $message)
+    {
+        $config    = new ConfigDouble();
+        $exception = 'PHP_CodeSniffer\Exceptions\DeepExitException';
+
+        if (method_exists($this, 'expectException') === true) {
+            // PHPUnit 5+.
+            $this->expectException($exception);
+            $this->expectExceptionMessage($message);
+        } else {
+            // PHPUnit 4.
+            $this->setExpectedException($exception, $message);
+        }
+
+        $config->processLongArgument($argument.'='.$value, 0);
+
+    }//end testInvalid()
+
+
+    /**
+     * Date provider for testInvalid()
+     *
+     * @see    self::testInvalid()
+     * @return array
+     */
+    public static function dataInvalidSniffs()
+    {
+        $arguments = [
+            'sniffs',
+            'exclude',
+        ];
+        $result    = [];
+
+        $sniffs = [];
+
+        $types = [
+            'Standard',
+            'Standard.Category',
+            'Standard.Category.Sniff.Code',
+        ];
+        foreach ($types as $value) {
+            $sniffs[$value] = $value;
+            $sniffs['Standard.Category.Sniff,B'.$value] = 'B'.$value;
+            foreach ($types as $extra) {
+                $sniffs['A'.$value.',B'.$extra] = 'A'.$value;
+            }
+        }
+
+        $messageTemplate = 'ERROR: The specified sniff code "%s" is invalid'.PHP_EOL.PHP_EOL;
+        foreach ($arguments as $argument) {
+            foreach ($sniffs as $input => $output) {
+                $result[] = [
+                    'argument' => $argument,
+                    'value'    => $input,
+                    'message'  => sprintf($messageTemplate, $output),
+                ];
+            }
+        }
+
+        return $result;
```

Comment:
As things are, there is quite some duplication in the test cases from what I can see. Not currently needed, but I can see the usecase for these tests with an eye on #344, so I'm fine with the duplication.